v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToBigQueryFlex.java

```diff
@@ -248,27 +252,74 @@ public static PipelineResult runAvroPipeline(
 
     WriteResult writeResult = null;
 
-    if (options.getAvroFormat().equals("NON_WIRE_FORMAT") && options.getAvroSchemaPath() != null) {
-
-      writeResult = kafkaRecords.apply(AvroTransform.of(options));
+    if (options.getMessageFormat().equals("AVRO_BINARY_ENCODING")
+        && options.getBinaryAvroSchemaPath() != null) {
+      writeResult =
+          kafkaRecords.apply(
+              AvroTransform.of(
+                  options.getReadBootstrapServerAndTopic().split(":")[2],
```

Comment:
nit: There is a check that verifies the format of this parameter in a different place, and then the same parameter is parsed again here. This will require modifying the code in 2 different places if the format changes. A better pattern would be to check the format and parse this option above, into 2 separate variables (bootstrapServers and topics), and then use these already parsed variables here.

Reply:
done